systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,112 @@
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String producerName = "producer-test";
+        final String consumerName = "consumer-test";
+        final String topicName = "topic-fg-cpl";
+        int sendingTime = 300;
+        List<EnvVar> testEnvVars = new ArrayList<>();
+        int kafkaReplicas = 2;
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+ControlPlaneListener", null));
+        deployClusterOperatorWithEnvVars(extensionContext, testEnvVars);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(NAMESPACE, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
+
+        LOGGER.info("Try to send some messages to Kafka over the next {} seconds.", sendingTime);
+        // use example clients for jobs - sending messages for 3-5mins
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBasicExampleClients.Builder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(sendingTime)
+                .withDelayMs(1000)
+                .withNamespaceName(NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi().build());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi().build());
+
+        LOGGER.info("Remove kafka-broker service and wait for it's recreation (should be done within 30secs)");
+        KubeClusterResource.kubeClient().deleteService(KafkaResources.brokersServiceName(clusterName));
+        ServiceUtils.waitForServiceDeletion(KafkaResources.brokersServiceName(clusterName), NAMESPACE);
+        ServiceUtils.waitForServiceCreation(KafkaResources.brokersServiceName(clusterName), NAMESPACE);
+
+        LOGGER.info("Force Rolling Update of Kafka by upscaling replicas to {}", kafkaReplicas + 2);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, k -> k.getSpec().getKafka().setReplicas(kafkaReplicas + 2), NAMESPACE);
+        StatefulSetUtils.waitForAllStatefulSetPodsReady(NAMESPACE, KafkaResources.kafkaStatefulSetName(clusterName), kafkaReplicas + 2, ResourceOperation.getTimeoutForResourceReadiness(Constants.STATEFUL_SET));
```

Comment:
This might do the job. But I wonder if this is the best solution. What about just deleting one of the broker pods?

Reply:
I can do both - delete 1 of 2 kafka pods before RU.

Reply:
Or just trigger manual RU via annotation?

Reply:
I am triggerring RU by changing KafkaReplicas from 2 -> 4.